File: Aula 13 - Análise discriminante/compra.R
# install.packages('readxl')
library(readxl)
setwd('D:/Users/25052288840/Downloads/FIA_data_mining/Aula 13 - Análise discriminante')
compra_xls <-read_excel('compra.xls.xlsx')

par(mfrow=c(1,3))
boxplot(durabilid~compra,
        data=compra_xls, main="durabilidade nos grupos", col=rainbow(2))
boxplot(desempenh~compra,
        data=compra_xls, main="desempenho nos grupos", col=rainbow(2))
boxplot(estilo~compra,
        data=compra_xls, main="estilo nos grupos", col=rainbow(2))
par(mfrow=c(1,1))

# install.packages('psych')
library(psych)
describe(compra_xls)

describeBy(compra_xls, compra_xls$compra)

var.test( durabilid~compra, data=compra_xls)
var.test( desempenh~compra, data=compra_xls)
var.test( estilo~compra, data=compra_xls)

t.test(var.equal=TRUE, durabilid~compra, data=compra_xls)
t.test(var.equal=FALSE, durabilid~compra, data=compra_xls)

t.test(var.equal=TRUE, desempenh~compra, data=compra_xls)

t.test(var.equal=TRUE, estilo~compra, data=compra_xls)
aov.res<-aov(durabilid~compra, data=compra_xls)
summary(aov.res)
aov.res<-aov(desempenh~compra, data=compra_xls)
summary(aov.res)
aov.res<-aov(estilo~compra, data=compra_xls)
summary(aov.res)

library(MASS)
result.lda<-lda(compra~., data=compra_xls)
predict(result.lda)

comprapredic<-read.csv("comprapredic.csv", header=TRUE, sep=";")
y_ = predict(result.lda, comprapredic)

print(y_)

install.packages('klaR')
library(klaR)
greedy.wilks(compra~., data=compra_xls, niveau=0.99)
greedy.wilks(compra~., data=compra_xls)

lda(compra~durabilid+desempenh, data=compra_xls)


lda(compra~durabilid, data=compra_xls)

plot(lda(compra~durabilid+desempenh, data=compra_xls))

# install.packages('car')
library(car)
qqPlot(compra_xls$durabilid, groups=compra_xls$compra)
qqPlot(compra_xls$desempenh, groups=compra_xls$compra)
qqPlot(compra_xls$estilo, groups=compra_xls$compra)


compraordem<-compra_xls[order(compra_xls$compra),]
compranao<-compraordem[compraordem$compra=="nao",]
comprasim<-compraordem[compraordem$compra=="sim",]

install.packages('nortest')
library(nortest)
lillie.test(compranao$durabilid)
shapiro.test(compranao$durabilid)
lillie.test(comprasim$durabilid)
shapiro.test(comprasim$durabilid)
lillie.test(compranao$desempenh)
shapiro.test(compranao$desempenh)
lillie.test(comprasim$desempenh)
shapiro.test(comprasim$desempenh)


install.packages('biotools')
library(biotools)
boxM(compra_xls[,2:4], compra_xls$compra)

File: Aula 13 - Análise discriminante/compra.py
import os
import pandas as pd

base_path = os.path.dirname(__file__)
compra_xls = pd.read_excel('compra.xls.xlsx')

compra_xls.boxplot(by='compra')

compra_xls.groupby('compra')['durabilid'].plot.hist()
compra_xls.groupby('compra')['estilo'].plot.hist()

print(compra_xls.describe())

compra_sim = compra_xls[compra_xls['compra'] == 'sim']
compra_nao = compra_xls[compra_xls['compra'] == 'nao']
print(compra_sim.describe())
print(compra_nao.describe())

import scipy.stats as stats
print(stats.f_oneway(
        compra_sim['desempenh'],
        compra_nao['desempenh'])
     )

print(stats.f_oneway(
        compra_sim['durabilid'],
        compra_nao['durabilid'])
     )

print(stats.f_oneway(
        compra_sim['estilo'],
        compra_nao['estilo'])
     )

from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
X = compra_xls[['durabilid', 'desempenh', 'estilo']]
y = compra_xls['compra'] == 'sim'
print(y)
clf = LinearDiscriminantAnalysis()
clf.fit(X, y)
print(clf.decision_function(X))
print(clf.score(X, y))

y_ = clf.predict(X)

print(clf)
print(clf.score(X, y))
print(clf.coef_, clf.intercept_)


comprapredic = pd.read_csv("comprapredic.csv", header=0, sep=";")
X2 = comprapredic[['durabilid', 'desempenh', 'estilo']]
clf.predict(X2)


from sklearn.feature_selection import RFE
clfrfe = LinearDiscriminantAnalysis()
selector = RFE(estimator=clfrfe)
selector.fit(X, y)
print(selector.support_)
print(selector.ranking_)
print(selector.feature_importances_)
Xrfe = selector.transform(X)


X_r = compra_xls[['durabilid', 'desempenh']]
clf2 = LinearDiscriminantAnalysis()
clf2.fit(X_r, y)
print(clf2.decision_function(X_r))
print(clf2.score(X_r, y))
print(clf2.coef_, clf2.intercept_)
